immuni_analytics/helpers/safety_net.py

```diff
@@ -0,0 +1,300 @@
+#  Copyright (C) 2020 Presidenza del Consiglio dei Ministri.
+#  Please refer to the AUTHORS file for more information.
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as
+#  published by the Free Software Foundation, either version 3 of the
+#  License, or (at your option) any later version.
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU Affero General Public License for more details.
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import base64
+import binascii
+import json
+import logging
+from datetime import datetime, timedelta
+from hashlib import sha256
+from json import JSONDecodeError
+from typing import Any, Dict, List
+
+import certvalidator
+import jwt
+from certvalidator import CertificateValidator
+from cryptography.exceptions import UnsupportedAlgorithm
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPublicKey
+from cryptography.x509 import Certificate, load_der_x509_certificate
+from jwt import DecodeError
+
+from immuni_analytics.core import config
+from immuni_analytics.models.operational_info import OperationalInfo
+from immuni_common.core.exceptions import ImmuniException
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_redis_key(salt: str) -> str:
+    """
+    Retrieve the redis key for a given salt.
+
+    :param salt: the salt corresponding to a request.
+    :return: the redis key containing the salt.
+    """
+    return f"~safetynet-used-salt:{salt}"
+
+
+class SafetyNetVerificationError(ImmuniException):
+    """Raised when one of the steps in the verification fails."""
+
+
+def _get_jws_part(jws_token: str, index: int) -> str:
```

Comment:
what about creating a helper which split the token in the 3 parts and return a named tuple? so that we don't repeat these splits over and over again in the functions below, also improving readability